src/functions/functions.jl

```diff
@@ -47,6 +47,27 @@ getfunction(f::FunctionWrapper) = f.f
 
 getdim(f::FunctionWrapper) = f.dim
 
+
+
+"""
+Wrapper for function whose first dim*dim input contains a matrix. 
+"""
+struct MatrixFunctionWrapper <: AbstractFunction
+    f::FunctionWrapper
```

Comment:
Is FunctionWrapper a concrete type, otherwise this field is not concrete

Reply:
It is. It's a function wrapper defined in Nonconvex.